### DISCUSSION TITLE
Consistent instructions for running as a container

### DISCUSSION THREAD

**Discussion by andrewheberle** [rustdesk/rustdesk-server-pro#294] [General]
There are two sets of instructions provided for running RustDesk Pro as a container as follows:

* https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/installscript/docker/
* https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/installscript/docker/rustdesk-server-layered-security-model/

The first set of instructions explicitly state you must set `network_mode: host` when using Docker Compose, while the other ignores this and sets a static hostname for the `hbbs` container instead.

Is it possible to confirm the correct requirements, especially in relation to licensing of the Pro version as I’ve experienced the issue where on reboot/recreate of the container the license is no longer accepted as valid, unless host mode networking is enabled.

If all that is required to maintain the license is a stable hostname of the container that would be great to know…especially if there was a desire to run RusDesk inside Kubernetes.

**Comment by rustdesk** [comment]
I suggest the first one.

The second one is provided by contributor, we have never used. And we have never used Kubernetes either.

**Reply by andrewheberle** [reply]
What is the license based on though?

If it’s based on the hostname that is fine, or is it based on the MAC address of the network interface?

The Windows instruction for Docker includes adding a “mac-address” option also. 

Just wanted to understand what the license takes into account to possibly enable running via Kubernetes etc

**Reply by rustdesk** [reply]
hostname and machine id

**Reply by andrewheberle** [reply]
Thanks for the clarification. So assuming the hostname of the container is stable and “/var/lib/dbus/machine-id” or “/etc/machine-id” is persistent also, it may be possible to not run using host mode on the container. 

So the container sees the real IP of the client we can disable the Docker userland proxy too. 

I’ve got an instance of RustDesk Pro set up for an evaluation at work so once we sort out a trial license I can report back so the Docker instructions might be more complete and avoid host mode networking.
